### DISCUSSION TITLE
In Gatsby 5, how to disable slices?

### DISCUSSION THREAD

**Discussion by gorohoroh** [gatsbyjs/gatsby#37481] [Help]
We've just upgraded from v4 to latest v5 in an attempt to improve performance.

I get it that v5 has slices. I'm currently not interested in using this API and I have performed no changes to make use of it.

Even though I'm not explicitly using slices, the `stiching slices` build step is taking from 20% to 50% of build time. I can live with that when I'm building with a small sample of data and it's taking 20 seconds, but when performing the full build, it's taking 50+ minutes for no apparent reason.

I have tried to launch my build with `GATSBY_SLICES=false gatsby build` and `GATSBY_SLICES=0 gatsby build`, but this doesn't make the `stiching slices` build step go away.

Is there a working way to opt out of this build step?

**Comment by LekoArts** [comment]
Hi!

You can't disable Slices in Gatsby 5, we use it to add scripts to the page: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/static-entry.js#L442

For my personal site (https://www.lekoarts.de/) which isn't small but also not super large the `stitching slices` step takes 0.13s. You're also the first person to report this, for any other site we haven't seen 50+ minute builds on this. The step normally takes under a second.

Can you create a https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/ ?

**Reply by gorohoroh** [reply]
Thanks.

The way I'm reading the piece of code that you're linking to, the created element only makes sense if the project has any user-defined slices. In my project, this is not the case.

In fact, [seeing how the env variable is defined](https://github.com/gatsbyjs/gatsby/blob/f8f084ae96bd105509903216daa766325ec65b9e/packages/gatsby/src/bootstrap/load-config/index.ts#L66), I was able to disable slices with the following snippet in `gatsby-node.js`:

`exports.onPostBootstrap = () => {
    delete process.env.GATSBY_SLICES;
};`

Comparing build artifacts with and without slices, I can see no meaningful difference in the generated HTML:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/507264/212955118-53158c83-4773-416c-91dc-6bf420c9106c.png">

Other than this, HTML is the same.

As to `page-data.json`, the only difference is whether a single property with an empty object (`"slicesMap": {}`) is generated.

Can you confirm that these differences are not substantial? Am I not taking any side effects into account?

**Reply by gorohoroh** [reply]
As to providing a reproducible test case, I'm afraid this is not an option. Providing access to source code could be an option, and if you like, we can discuss this on a call with your Account Executive later this week. Really struggling with Gatsby performance on all fronts with this project.

**Reply by LekoArts** [reply]
> the created element only makes sense if the project has any user-defined slices

No, it's to add code to the site that every site needs.

> Can you confirm that these differences are not substantial? Am I not taking any side effects into account?

Your site rebuilds will be slower since the slice that gets updated won't be stitched into other pages. The whole reason why we use slices for that snippet is to speed up builds. We're also relying on it being enabled/available so I can't guarantee that things won't break in the future in that regard since we consider it always enabled (and we'll most likely remove the environment variable).

> As to providing a reproducible test case, I'm afraid this is not an option

Why not? https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/ explains that you should create a new project that shows the behavior, not share the source code of your production site. Because if you can't reproduce it in a minimal project it means your environment is at fault.

> Really struggling with Gatsby performance on all fronts with this project

Where are you building the site?
